api/app/helpers/transcript.ts

```diff
@@ -21,4 +24,53 @@ function generateUniqueHash(content: any) {
   return buffer.toString("base64");
 }
 
-export { generateUniqueStr, generateUniqueHash };
+function parseMdToJSON<T extends BaseParsedMdContent>(mdContent: string): T {
+  const regex = /^---\n([\s\S]*?)\n---\n([\s\S]*)$/;
+  const match = mdContent.match(regex);
+
+  if (!match) {
+    throw new Error("Invalid Markdown format");
+  }
+
+  const [, header, body] = match;
+
+  const json: Record<string, string | string[] | undefined> = {} as Partial<T>;
+  const lines = header.split("\n");
+
+  for (const line of lines) {
+    const [key, ...valueParts] = line.split(":");
+    if (valueParts.length > 0) {
+      const value = valueParts.join(":").trim();
+      if (key.trim() !== "") {
+        if (value.startsWith("[") && value.endsWith("]")) {
+          try {
+            // Replace single quotes with double quotes before parsing
+            const validJson = value.replace(/'/g, '"');
+            json[key] = JSON.parse(validJson);
+          } catch (error) {
+            throw new Error("Invalid JSON format");
+          }
+        } else {
+          // Handle other key-value pairs
+          if (value.startsWith("'") && value.endsWith("'")) {
+            json[key] = value.slice(1, -1);
+          } else if (value.startsWith("'") && value.endsWith("',")) {
+            if (!json[key]) {
+              json[key] = [];
+            }
+            (json[key] as string[]).push(value.slice(1, -2));
+          } else {
+            json[key] = value;
+          }
+        }
+      }
+    }
+  }
```

Comment:
The "header" of the markdown file is in YAML format. So you can avoid all this custom parsing logic if you use the `js-yaml` package.

This custom logic is also not taking into consideration all the "edge" cases. So having a proper YAML parser makes sense.

I just tested it and it seems that you can replace all this custom logic with
```suggestion
import yaml from "js-yaml"
....
const json: Record<string, string | string[] | undefined> = yaml.load(header) as Partial<T>
```

Reply:
Cool, was not sure what format tstbtc was sending in the first place hence why the custom parser.